aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/CloudWatchLogsClientSpec.groovy

```diff
@@ -1,24 +1,40 @@
 package io.micronaut.aws.sdk.v2.service
 
-import io.micronaut.aws.sdk.v2.ApplicationContextSpecification
+
+import software.amazon.awssdk.core.client.config.SdkClientOption
 import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient
 import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient
 
-class CloudWatchLogsClientSpec extends ApplicationContextSpecification {
+class CloudWatchLogsClientSpec extends ServiceClientSpec {
+
+    @Override
+    protected String serviceName() {
+        return CloudWatchLogsClient.SERVICE_NAME
+    }
 
     void "it can configure a sync client"() {
         when:
         CloudWatchLogsClient client = applicationContext.getBean(CloudWatchLogsClient)
 
         then:
-        client.serviceName() == CloudWatchLogsClient.SERVICE_NAME
+        client.serviceName() == serviceName()
     }
 
     void "it can configure an async client"() {
         when:
         CloudWatchLogsAsyncClient client = applicationContext.getBean(CloudWatchLogsAsyncClient)
 
         then:
-        client.serviceName() == CloudWatchLogsClient.SERVICE_NAME
+        client.serviceName() == serviceName()
+    }
+
+    void "it can override the endpoint"() {
```

Comment:
Perhaps you could move this test to `ServiceClientSpec` and add abstract methods for the subclasses to provide the actual `SdkClient` client and asyncClient instances.